### PR TITLE
feat(ui5-date-picker): add screen reader support

### PR DESCRIPTION
**Pull request by Todor-ads**
Fixes: https://github.com/SAP/ui5-webcomponents/issues/1279

### CODE REVIEW

File: packages/main/src/DayPicker.hbs [hunk 1]
```diff
@@ -48,12 +40,20 @@
...
 							</div>
-						{{/each}}
-					</div>
-				{{else}}
-					<div class="sapWCEmptyWeek"></div>
-				{{/if}}
-			{{/each}}
-		</div>
+						{{else}}
+							{{#unless _hideWeekNumbers}}
```
Comment (tsanislavgatev) [inline packages/main/src/DayPicker.hbs:44]: After this change the hideWeeknumbers property does no longer work.

File: packages/main/src/DayPicker.js [hunk 2]
```diff
@@ -338,6 +340,10 @@ class DayPicker extends UI5Element {
 		let dayName;
 
 		this._dayNames = [];
+		this._dayNames.push({
+			classes: "sapUiCalDummy ui5-dp-dayname",
```
Comment (tsanislavgatev) [inline packages/main/src/DayPicker.js:344]: sapUiCalDummy?

File: packages/main/test/specs/DatePicker.spec.js [hunk 3]
```diff
@@ -400,7 +400,7 @@ describe("Date Picker Tests", () => {
 		datepicker.valueHelpIcon.click();
 
 		datepicker.btnPrev.click();
-
+		// browser.debug()
```
Comment (tsanislavgatev) [inline packages/main/test/specs/DatePicker.spec.js:403]: Please remove

File: packages/main/test/specs/DatePicker.spec.js [hunk 4]
```diff
@@ -718,6 +718,77 @@ describe("Date Picker Tests", () => {
...
+
+	it("DatePcker dates and week number", ()=>{
+		browser.url("http://localhost:8080/test-resources/pages/DatePicker_test_page.html?sap-ui-language=en");
+		datepicker.root.setAttribute("primary-calendar-type", "Gregorian");
+		datepicker.id = "#dp13";
+		datepicker.openPicker({ focusInput: true });
+		datepicker.root.keys("May 3, 2100");
+		datepicker.root.keys("Enter");
+
+		const data = Array.from(datepicker.getDayPickerDatesRow(2));
```
Comment (tsanislavgatev) [inline packages/main/test/specs/DatePicker.spec.js:785]: Can we have strictEqual for the tests?

File: packages/main/test/pageobjects/DatePickerTestPage.js [hunk 5]
```diff
@@ -99,8 +99,28 @@ class DatePickerTestPage {
...
+		return dayNames[0].$$("div");
+	}
+	
+	getDayPickerDatesRow(index) {
+		const data = Array.from(this.getDayPickerContent());
+		return data[index].$$("div");
+	}
+
+	getDayPickerNumbers() {
+		const dayNames = Array.from(this.getDayPickerContent());
```
Comment (tsanislavgatev) [inline packages/main/test/pageobjects/DatePickerTestPage.js:121]: Why don't just return this?

**Review by tsanislavgatev — CHANGES_REQUESTED** [review]

File: packages/main/test/specs/DatePicker.spec.js (hunk 3 above)
Comment (Todor-ads) [reply packages/main/test/specs/DatePicker.spec.js:403]: Done

File: packages/main/src/DayPicker.js (hunk 2 above)
Comment (Todor-ads) [reply packages/main/src/DayPicker.js:344]: Done

File: packages/main/src/CalendarHeader.js [hunk 6]
```diff
@@ -123,6 +134,22 @@ class CalendarHeader extends UI5Element {
...
+
+	get _prevButtonText() {
+		return this.i18nBundle.getText(CALENDAR_HEADER_PREVIOUS_BUTTON);
+	}
+
+	get _nextButtonText() {
+		return this.i18nBundle.getText(CALENDAR_HEADER_NEXT_BUTTON);
+	}
+
+	get _fullMonthName() {
```
[the remark this answers is not in the document]
Comment (Todor-ads) [reply packages/main/src/CalendarHeader.js:150]: Done

File: packages/main/src/CalendarHeader.js [hunk 7]
```diff
@@ -81,6 +88,10 @@ class CalendarHeader extends UI5Element {
 
 		this._btn2 = {};
 		this._btn2.type = ButtonDesign.Transparent;
+
+		this._oLocale = getLocale();
```
[the remark this answers is not in the document]
Comment (Todor-ads) [reply packages/main/src/CalendarHeader.js:92]: Done

File: packages/main/test/specs/DatePicker.spec.js (hunk 4 above)
Comment (Todor-ads) [reply packages/main/test/specs/DatePicker.spec.js:785]: Done

File: packages/main/test/pageobjects/DatePickerTestPage.js (hunk 5 above)
Comment (Todor-ads) [reply packages/main/test/pageobjects/DatePickerTestPage.js:121]: Done

**Review by ilhan007 — COMMENTED** [review]
The following DatePicker test is failing "✖ Tests week numbers column visibility".
You can find it in the build output, following the "continuous-integration/travis-ci/pr" link at the bottom of the Pull Request page.

**Review by tsanislavgatev — CHANGES_REQUESTED** [review]

File: packages/main/src/DayPicker.hbs (hunk 1 above)
Comment (Todor-ads) [reply packages/main/src/DayPicker.hbs]: done

File: packages/main/test/specs/DatePicker.spec.js [hunk 8]
```diff
@@ -718,6 +718,77 @@ describe("Date Picker Tests", () => {
...
+	it("DayPiker day number attribute", ()=>{
+		browser.url("http://localhost:8080/test-resources/pages/DatePicker_test_page.html?sap-ui-language=en");
+		datepicker.root.setAttribute("primary-calendar-type", "Gregorian");
+		datepicker.id = "#dp13";
+		datepicker.openPicker({ focusInput: true });
+		datepicker.root.keys("May 3, 2100");
+		datepicker.root.keys("Enter");
+
+		const content = Array.from(datepicker.getDayPickerNumbers());
+		for(let i = 1; i < content.length; i++){
```
[the remark this answers is not in the document]
Comment (Todor-ads) [reply packages/main/test/specs/DatePicker.spec.js]: Done

File: packages/main/src/DayPicker.js [hunk 9]
```diff
@@ -243,6 +241,8 @@ class DayPicker extends UI5Element {
 		let week = [];
 		this._weekNumbers = [];
 		let weekday;
+		const _aMonthsNameWide = this._oLocaleData.getMonths("wide", this._calendarDate._oUDate.sCalendarType);
+
 		if (this.minDate) {
```
[the remark this answers is not in the document]
Comment (Todor-ads) [reply packages/main/src/DayPicker.js:246]: done

File: packages/main/test/specs/DatePicker.spec.js [hunk 10]
```diff
@@ -696,9 +696,9 @@ describe("Date Picker Tests", () => {
 		// act
 		datepicker.id = "#dp18";
 		datepicker.valueHelpIcon.click()
-
+
```
[the remark this answers is not in the document]
Comment (Todor-ads) [reply packages/main/test/specs/DatePicker.spec.js:699]: Done

File: packages/main/src/DayPicker.js [hunk 11]
```diff
@@ -260,6 +260,9 @@ class DayPicker extends UI5Element {
 			if (weekday < 0) {
 				weekday += 7;
 			}
+
+			const nonWorkingAriaLabel = this._isWeekend(oCalDate) ? "Non-Working Day " : "";
```
Comment (ilhan007) [inline packages/main/src/DayPicker.js:264]: Shouldn't be this translated as well "Non-Working Day"

**Review by ilhan007 — CHANGES_REQUESTED** [review]

File: packages/main/src/DayPicker.js (hunk 11 above)
Comment (Todor-ads) [reply packages/main/src/DayPicker.js:264]: Done

**Comment by ilhan007** [comment]
The changes are merged with another PR: https://github.com/SAP/ui5-webcomponents/pull/2224

**Pull request closed without merging**